src/python/WMCore/Services/pycurl_manager.py

```diff
@@ -78,6 +79,10 @@ def set_opts(self, curl, url, params, headers,
         curl.setopt(pycurl.FOLLOWLOCATION, self.followlocation)
         curl.setopt(pycurl.MAXREDIRS, self.maxredirs)
 
+        if params:
+            if  isinstance(params, dict):
+                params = json.dumps(params)
+
         if  verb == 'GET':
             encoded_data = urllib.urlencode(params, doseq=doseq)
```

Comment:
I don't think I should care about string content. I add the code which should convert dict to proper json stream. But if user will pass wrong parameter the code should not do the validation here, it is up to the client.